### PR TITLE
Dev/llm info extract

### PR DESCRIPTION
**Pull request by BeachWang**
Add information extraction OPs, including:
1. text_chunk_mapper: split text into chunks.
2. extract_entity_attribute_mapper: extract given attribute from the text.
3. extract_entity_relation_mapper: modified from lightRAG, extract entities and relations for knowledge graph from text.
4. extract_event_mapper: extract events and relevant characters from text.
5. extract_keyword_mapper: extract keywords from text, prompt is modified from lightRAG.
6. extract_nickname_mapper: extract nicknames from the text.

Minor fix:
1. add retry_num for parsing fail for LLM API calling.
2. default model_params and sampling_params to be `{}`, since `**` operator is invalid for None.

### CODE REVIEW

File: data_juicer/utils/common_utils.py [hunk 1]
```diff
@@ -19,3 +19,11 @@ def stats_to_number(s, reverse=True):
             return -sys.maxsize
         else:
             return sys.maxsize
+
+
+def is_float(s):
+    try:
+        float(s)
+        return True
+    except ValueError:
```
Comment (drcege) [inline data_juicer/utils/common_utils.py:28]: 也会出现 TypeError，比如 `float(None)`。 可以直接用 Exception

File: data_juicer/ops/mapper/extract_entity_relation_mapper.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+            messages = self.add_message(history, 'user', self.if_loop_prompt)
+            if_loop_result = client(messages, **self.sampling_params)
+            if_loop_result = if_loop_result.strip().strip('"').strip(
+                "'").lower()
+            if if_loop_result != 'yes':
+                break
+
+        return final_result
+
+    def process_single(self, sample=None, rank=None):
```
Comment (drcege) [inline data_juicer/ops/mapper/extract_entity_relation_mapper.py:318]: 没有必要 `sample=None`

File: data_juicer/ops/mapper/extract_keyword_mapper.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        output_pattern = re.compile(self.output_pattern,
+                                    re.VERBOSE | re.DOTALL)
+        matches = output_pattern.findall(raw_output)
+        for record in matches:
+            items = split_text_by_punctuation(record)
+            keywords.append(items)
+
+        return keywords
+
+    def process_single(self, sample=None, rank=None):
```
Comment (drcege) [inline data_juicer/ops/mapper/extract_keyword_mapper.py:167]: sample=None -> sample

File: data_juicer/ops/mapper/extract_nickname_mapper.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+            nickname = nickname.strip()
+            # is name but not nickname
+            if role2 == nickname:
+                continue
+            if role1 and role2 and nickname:
+                nickname_relations.append((role1, role2, nickname))
+        nickname_relations = list(set(nickname_relations))
+
+        nickname_relations = [{
+            'source_entity': nr[0],
```
Comment (drcege) [inline data_juicer/ops/mapper/extract_nickname_mapper.py:125]: 这些 key 是固定的吗？

Comment (BeachWang) [reply data_juicer/ops/mapper/extract_nickname_mapper.py:125]: 固定的，根据lightRAG设置的。

Comment (BeachWang) [reply data_juicer/ops/mapper/extract_nickname_mapper.py]: 放到constant.py里了

**Review by drcege — APPROVED** [review]

**Pull request merged**